### PR TITLE
Fix not signing proof + add proof to essential schema

### PR DESCRIPTION
**Pull request by SamHellawell**
The proof object was only being applied in toJSON which meant it wasnt being signed, especially noticeable with SDK credentials

### CODE REVIEW

File: src/anonymous-credentials/credential-builder.ts [hunk 1]
```diff
@@ -167,15 +170,15 @@ export class CredentialBuilder extends Versioned {
       s[STATUS_STR] = this._credStatus;
     }
 
+    if (!s['proof']) {
```
Comment (lovesh) [inline src/anonymous-credentials/credential-builder.ts:173]: Proof is generated after signing so why is the "type" of proof being included in what's supposed to be signed? Or are there other details about the proof that need to be known before signing?

File: src/anonymous-credentials/credential.ts [hunk 2]
```diff
@@ -12,6 +12,13 @@ import { VerifyResult } from '@docknetwork/crypto-wasm';
 import { verifyMessageObject } from '../sign-verify-js-objs';
 import b58 from 'bs58';
 
+function isEmptyObject(obj) {
```
Comment (lovesh) [inline src/anonymous-credentials/credential.ts:15]: This is a utility function so should be moved to an appropriate file

File: src/anonymous-credentials/credential-builder.ts (hunk 1 above)
Comment (SamHellawell) [reply src/anonymous-credentials/credential-builder.ts:173]: in the SDK there are other details required in proof. i would also argue that its important to sign each field in this lib's credential impl, including its type. probably makes sense to include a verification method and/or proof created dates in this impl too for signing (the SDK BBS+ credentials already have these fields and more)

Comment (lovesh) [reply src/anonymous-credentials/credential-builder.ts:173]: Ok.

**Pull request merged**